src/Components/Hints.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { BsCircleFill, BsXCircle, BsCircle } from "react-icons/bs";
+import { constraints } from "../constants";
+
+const Hints = () => {
+  const [show, setShow] = useState<boolean>(false);
+
+  return (
+    <>
+      <div className="hints">
+        <h4
+          onClick={() => {
+            setShow((show) => !show);
```

Comment:
there is conflict in name here
`show` is the name of state as well and `show` is the name variable in this function also

Reply:
changed the name show to prev